src/App.js

```diff
@@ -1,56 +1,203 @@
-import './Components/DynamicContent/Config/Config.css';
 import './Components/DynamicContent/Asset/AssetManager.css';
+import './Components/DynamicContent/PromotionBuilder/PromotionBuilder.css';
 import './Components/DynamicContent/Container.css';
 import './App.css';
+import { promotionData } from './Components/DynamicContent/PromotionBuilder/promotionData';
+import { getTruthyList , arrayToMap } from './utils/utils'
 
 
 class App extends Component {
 
-  state = { 
-    view: "Search"
+  state = {
+    view: "Search",
+    selectedPromotionId: null,
+    formData: {
+      id: 0,
+      devices: new Map(),
+      ventures: new Map(),
+      url: "",
+      name: "",
+      lastUpdatedTime: ""
+    },
+    searchTerm: "",
+    searchDisplay: [],
+    promotionData: promotionData
   };
-  
-  onclick = (event) => {
+
+
+  componentDidMount(){
+    this.searchPromotions();
+  }
+
+
+  componentDidUpdate(prevProps, prevState){
+    if(!isEqual(prevState.promotionData,this.state.promotionData)){
+      this.searchPromotions();
+    }
+    
+  }
+
+  handleInputChange = event => this.setState({ searchTerm: event.target.value });
+
+
+  searchBtnClick = event => {
+    event.preventDefault();
+    this.searchPromotions();
+  }
+
+
+  searchPromotions = () => {
+    const searchInput = this.state.searchTerm;
+    const promotionArray = Object.keys(this.state.promotionData).map(key => this.state.promotionData[key]);
+    const filteredPromotionArray = promotionArray.filter((el) => el.name.toLowerCase().indexOf(searchInput.toLowerCase()) > -1);
+    this.setState({searchDisplay: filteredPromotionArray.sort((a,b) => b.lastUpdatedTime - a.lastUpdatedTime) });
+  }
+
+
+  onFormChange = (value, formField) => {
+      this.setState(prevState => {return {  formData: { ...prevState.formData, [formField]: value }}})
+  }
+
+
+  savePromotion = () => {
+    if (this.state.formData) {
+        let id = this.getPromotionId();
+        this.showAlertWithFormData();
+        this.setState(prevState => {return {  promotionData: { ...prevState.promotionData, [id ]: this.getNewOrUpdatedPromotion(id)}}})
+        this.setState({view: "Search"})
+    }
+  }
+
+
+  getNewOrUpdatedPromotion = (id) => {
+
+    let newPromotion = {
+      id: null,
+      devices: [],
+      ventures: [],
+      url: "",
+      name: "",
+      lastUpdatedTime: ""
+    }
+
+    newPromotion.id=id;
+    newPromotion.name=this.state.formData.name;
+    newPromotion.url=this.state.formData.url;
+    newPromotion.devices=[...this.state.formData.devices.keys()]
+    newPromotion.ventures=[...this.state.formData.ventures.keys()]
+    newPromotion.lastUpdatedTime=Date.now()
+
+    return newPromotion;
+  }
+
+
+  getPromotionId = () => {
+
+    let uuid4 = require('uuid4');
+    let id = null;
+    if(this.state.selectedPromotionId){
+      id = this.state.selectedPromotionId
+    }
+
+    if(this.state.formData.id){
+        id = this.state.formData.id
+    } else {
+      id = uuid4();
+    }
```

Comment:
this can be simplified using destructuring. Consider the following:
```
const {
  selectedPromotionId,
  formData: {
    id: formDataId
  }
} = this.state;

return formDataId || selectedPromotionId || uuid4();
```